src/web/mjs/connectors/TuMangaOnlineHentai.mjs

```diff
@@ -22,81 +22,38 @@ export default class TuMangaOnlineHentai extends Connector {
         return new Manga(this, id, title);
     }
 
-    /**
-     *
-     */
-    _getMangaListFromPages( mangaPageLinks, index ) {
-        index = index || 0;
-        let request = new Request( mangaPageLinks[ index ], this.requestOptions );
-        return this.fetchDOM( request, 'div.panel-body table.table tbody tr td.text-left a', 5 )
-            .then( data => {
-                let mangaList = data.map( element => {
-                    return {
-                        id: element.href.split( '/' ).pop(),
-                        title: element.text.trim()
-                    };
-                } );
-                if( index < mangaPageLinks.length - 1 ) {
-                    return this._getMangaListFromPages( mangaPageLinks, index + 1 )
-                        .then( mangas => mangaList.concat( mangas ) );
-                } else {
-                    return Promise.resolve( mangaList );
-                }
-            } );
+    async _getMangas() {
+        let mangaList = [];
+        let hasNext = true;
+        let page = 1;
+        while (hasNext) {
```

Comment:
May use [for loop](https://github.com/manga-download/hakuneko/blob/d0452f8047446166f68220c9f656e6b6adfc1c14/src/web/mjs/connectors/templates/%40Samples.mjs#L50) `for(let page = 1, hasNext = true; hasNext; page++)`